pkg/nfd-master/nfd-master.go

```diff
@@ -229,7 +246,22 @@ func filterFeatureLabels(labels Labels, extraLabelNs []string, labelWhiteList *r
 			delete(labels, label)
 		}
 	}
-	return labels
+
+	// Remove labels which are intended to be extended resources
+	extendedResources := ExtendedResources{}
+	for _, extendedResourceName := range extendedResourceNames {
+		if _, ok := labels[extendedResourceName]; ok {
+			if _, err := strconv.Atoi(labels[extendedResourceName]); err != nil {
+				stderrLogger.Printf("bad label value encountered for extended resource: %s", err.Error())
+				continue // non-numeric label can't be used
+			}
+
+			extendedResources[extendedResourceName] = labels[extendedResourceName]
+			delete(labels, extendedResourceName)
+		}
+	}
```

Comment:
I think we should do label namespace/prefix handling here, dropping the default namespace if specified (i.e. something like `strings.TrimPrefix(extendedResourceName , LabelNs)`). This way, the use could specify labels in the default ns in two ways, either in the fully namespaced format (`feature.node.kubernetes.io/source-feature.name`) or in a shorter (`source-feature.name`) format. This would comply with the label handling/naming used in the `local` feature source in nfd-worker.

This also needs to be documented in README